sharedbuffers/mapped_struct.py

```diff
@@ -4673,13 +5058,43 @@ def _py__hinted_bsearch(a, hkey, hint, lo, hi, check_equal):
     globals()['_hinted_bsearch'] = _py__hinted_bsearch
 
     def _py_hinted_bsearch(a, hkey, hint):
+        """
+        Search into the sorted array ``a`` the value ``hkey``, assuming it should
+        be close to ``hint``.
+
+        :param ndarray a: Sorted array of one of the supported types (see :func:`bsearch`).
+
+        :param hkey: Value to search for
+
+        :param int hint: Expected location of ``hkey``
+
+        :rtype: int
+        :returns: Location where ``hkey`` should be, if it is there. See :func:`bsearch`.
+        """
         return bisect.bisect_left(a, hkey)
+    _py_hinted_bsearch.__name__ = 'hinted_bsearch'
     globals()['hinted_bsearch'] = _py_hinted_bsearch
 
 #@cython.ccall
 @cython.locals(lo = cython.size_t, hi = cython.size_t)
 #@cython.returns(cython.size_t)
 def bsearch(a, hkey):
+    """
+    Search into the sorted array ``a`` the value ``hkey``.
+
+    :param ndarray a: Sorted array of one of the supported types:
+
+        * ints of 8, 16, 32 and 64 bits, signed and unsigned
```

Comment:
I don't know. Habit. Sometimes rst doesn't work well without the blank lines, sometimes it does, I have never learned why so I put the blank line just in case when I remember to.